### PR TITLE
Work fe2 marin json protocol for add transaction

### PR DESCRIPTION
**Pull request by zoemarin**
Add all the required files for the json schema

### CODE REVIEW

**Review by lhmerino — DISMISSED** [review]

**Review by Tesa27 — CHANGES_REQUESTED** [review]
Lot of code in this PR is present in [#904](https://github.com/dedis/popstellar/pull/904#pullrequestreview-931386399) I think that the issues should be first address in that PR before this one

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/digitalcash/Transaction.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    public long getTimestamp() {
+        return timestamp;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Transaction that = (Transaction) o;
+        //TODO : Problem by check equality of two list ?
```
Comment (ahominal) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/digitalcash/Transaction.java:60]: Order of elements of the list to compare should be equal? List(1, 2, 3) != List(2, 1, 3)

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/digitalcash/TxIn.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    public ScriptTxIn getScript() {
+        return script;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        TxIn txIn = (TxIn) o;
```
Comment (ahominal) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/digitalcash/TxIn.java:46]: check if o.isInstanceOf(TxIn.class) before cast?

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/digitalcash/TxOut.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+  public ScriptTxOut getScript() {
+    return script;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+    TxOut txOut = (TxOut) o;
```
Comment (ahominal) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/digitalcash/TxOut.java:36]: same as above

**Review by ahominal — COMMENTED** [review]
Good work!

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/digitalcash/TxIn.java (hunk 2 above)
Comment (zoemarin) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/digitalcash/TxIn.java:46]: getClass() != o.getClass() does is it is not done with this ?

**Pull request closed without merging**